calcite-adapter/src/main/java/software/amazon/documentdb/jdbc/metadata/DocumentDbTableSchemaGenerator.java

```diff
@@ -324,7 +324,7 @@ private static void processArray(
             handleComplexScalarConflict(tableMap, tableName, columnMap);
         }
 
-        if (!tableMap.containsKey(path)) {
+        if (!tableMap.containsKey(tableName)) {
```

Comment:
changed this check to match the rest - the code in here was always being executed since tableMap uses the table names as the key